packages/@n8n/task-runner/src/js-task-runner/__tests__/js-task-runner.test.ts

```diff
@@ -1342,4 +1342,22 @@ describe('JsTaskRunner', () => {
 			task.cleanup();
 		});
 	});
+
+	describe('prototype pollution prevention', () => {
+		test('should prevent Object.setPrototypeOf in runOnceForAllItems', async () => {
+			const outcome = await executeForAllItems({
+				code: 'Object.setPrototypeOf({}, {}); return [{json: {result: true}}]',
+				inputItems: [{ a: 1 }],
+			});
+			expect(outcome.result).toEqual([wrapIntoJson({ result: true })]);
```

Comment:
This doesn't seem to actually test anything? The code would work exactly the same if `Object.setPrototypeOf({}, {})` is removed, right?

Reply:
Let me find a better way to test this

Reply:
[`c5f26d4`](https://github.com/n8n-io/n8n/pull/12588/commits/c5f26d4cabd007aad5086274a56cf65dfb5abe8b)